공장에서는 진공관 수명이 1만 시간이라고 주장하여 품질관리팀에서 12개 샘플을 뽑았음 유의수준 5%에서 부호 검정하시오.

In [23]:
import pandas as pd
import numpy as np
data=pd.read_csv("problem2.csv")

In [24]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   name       12 non-null     object
 1   life span  12 non-null     int64 
dtypes: int64(1), object(1)
memory usage: 320.0+ bytes


In [25]:
data

name  life span
0    sample1      10000
1    sample2       9000
2    sample3       9500
3    sample4      10000
4    sample5      10000
5    sample6       8900
6    sample7       9900
7    sample8      10100
8    sample9      10300
9   sample10      10020
10  sample11      10000
11  sample12       9800

H0 : 진공관 평균 수명시간은 1만시간이다.\
H1 : 진공관 평균 수명시간은 1만시간이 아니다.

2-(2) 유효한 데이터 개수는?

In [26]:
data.median()

C:\Users\Public\Documents\ESTsoft\CreatorTemp/ipykernel_34920/4184645713.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  data.median()


life span    10000.0
dtype: float64

In [21]:
#중위수와 동일한 값들은 부호 검정에서 불필요한 데이터이다.
data_filter=data[data['life span']!=10000]

In [15]:
data_filter

name  life span
1    sample2       9000
2    sample3       9500
5    sample6       8900
6    sample7       9900
7    sample8      10100
8    sample9      10300
9   sample10      10020
11  sample12       9800

In [29]:
len(data[data['life span']==10000])

4

In [31]:
import scipy.stats as stats
stats.wilcoxon(data['life span']-10000)

C:\Users\taeeu\anaconda3\lib\site-packages\scipy\stats\morestats.py:3141: UserWarning: Exact p-value calculation does not work if there are ties. Switching to normal approximation.
  warnings.warn("Exact p-value calculation does not work if there are "
C:\Users\taeeu\anaconda3\lib\site-packages\scipy\stats\morestats.py:3155: UserWarning: Sample size too small for normal approximation.
  warnings.warn("Sample size too small for normal approximation.")


WilcoxonResult(statistic=8.5, pvalue=0.18289327757982765)

검정통계량은 8.5이고 p-value는 0.05보다 크므로 귀무가설을 기각할 수 없다. 따라서 진공관 평균수명시간은 1만시간이라고 할 수 있다.